pyvista/core/filters/image_data.py

```diff
@@ -946,3 +949,314 @@ def contour_labeled(
         # Restore the original vtkLogger verbosity level
         _vtk.vtkLogger.SetStderrVerbosity(verbosity)
         return cast(pyvista.PolyData, wrap(alg.GetOutput()))
+
+    def pad_image(
+        self,
+        value: Union[float, VectorLike[float], Literal['wrap', 'mirror']] = 0.0,
+        *,
+        pad_size: int | VectorLike[int] = 1,
+        all_dimensions: bool = False,
+        scalars: Optional[str] = None,
+        all_scalars: bool = False,
+        progress_bar=False,
+    ) -> pyvista.ImageData:
+        """Enlarge an image by padding its boundaries with new points.
+
+        Padded point data may be mirrored, wrapped, or filled with a constant value.
+        By default, all boundaries of the image are padded with a single constant value.
+
+        This filter is designed to work with 1D, 2D, or 3D image data and will only pad
+        non-singleton dimensions unless otherwise specified.
+
+        Parameters
+        ----------
+        value : float | sequence[float] | 'mirror' | 'wrap', default : 0.0
```

Comment:
Not sure if there's a better name for this than `value`. Maybe `pad_value` to be consistent with `pad_size`? `pad_mode` probably best describes `'wrap'` and `'mirror'` though...